src/renderer/extensions/vueNodes/widgets/components/WidgetSelectButton.test.ts

```diff
@@ -0,0 +1,426 @@
+        { label: 'Second Option', value: 'second' },
+        { label: 'Third Option', value: 'third' }
+      ]
+      const widget = createMockWidget('second', { values: options })
+      const wrapper = mountComponent(widget, 'second')
+
+      const buttons = wrapper.findAll('button')
+      expect(buttons[0].text()).toBe('First Option')
+      expect(buttons[1].text()).toBe('Second Option')
+      expect(buttons[2].text()).toBe('Third Option')
+
+      // 'second' should be selected
+      expect(buttons[1].classes()).toContain('bg-white')
+    })
+
+    it('emits correct values for object options', async () => {
+      const options = [
+        { label: 'First', value: 'first_val' },
+        { label: 'Second', value: 'second_val' }
+      ]
+      const widget = createMockWidget('first_val', { values: options })
+      const wrapper = mountComponent(widget, 'first_val')
+
+      await clickSelectButton(wrapper, 'Second')
+
+      const emitted = wrapper.emitted('update:modelValue')
+      expect(emitted).toBeDefined()
+      expect(emitted![0]).toEqual(['second_val'])
+    })
+  })
+
+  describe('Edge Cases', () => {
+    it('handles options with special characters', () => {
+      const options = ['@#$%^&*()', '{}[]|\\:";\'<>?,./']
+      const widget = createMockWidget(options[0], { values: options })
+      const wrapper = mountComponent(widget, options[0])
+
+      const buttons = wrapper.findAll('button')
+      expect(buttons[0].text()).toBe('@#$%^&*()')
+      expect(buttons[1].text()).toBe('{}[]|\\:";\'<>?,./')
+    })
+
+    it('handles empty string options', () => {
+      const options = ['', 'not empty', '  ', 'normal']
+      const widget = createMockWidget('', { values: options })
+      const wrapper = mountComponent(widget, '')
+
+      const buttons = wrapper.findAll('button')
+      expect(buttons).toHaveLength(4)
+      expect(buttons[0].classes()).toContain('bg-white') // Empty string is selected
+    })
+
+    it('handles null/undefined in options', () => {
+      const options = ['valid', null, undefined, 'another'] as any[]
+      const widget = createMockWidget('valid', { values: options })
+      const wrapper = mountComponent(widget, 'valid')
+
+      const buttons = wrapper.findAll('button')
+      expect(buttons).toHaveLength(4)
+      expect(buttons[0].classes()).toContain('bg-white')
+    })
+
+    it('handles very long option text', () => {
+      const longText =
+        'This is a very long option text that might cause layout issues if not handled properly'
+      const options = ['short', longText, 'normal']
+      const widget = createMockWidget('short', { values: options })
+      const wrapper = mountComponent(widget, 'short')
+
+      const buttons = wrapper.findAll('button')
+      expect(buttons[1].text()).toBe(longText)
+    })
+
+    it('handles large number of options', () => {
+      const options = Array.from({ length: 20 }, (_, i) => `option${i + 1}`)
```

Comment:
If you didn't do the `+ 1`, you wouldn't have to explain the different label/index thing